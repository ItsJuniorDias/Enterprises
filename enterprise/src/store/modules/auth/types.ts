export enum ActionTypes {
  loginAuthRequest = 'LOGIN_AUTH_REQUEST',
  loginAuthSuccess = 'LOGIN_AUTH_SUCCESS',
  loginAuthFailure = 'LOGIN_AUTH_FAILURE',
  logoutUser = 'LOGOUT_USER',
}

export interface IInvestor {
  id: number;
  investor_name: string;
  email: string;
  city: string;
  country: string;
  balance: number;
  photo: any;
  portfolio: object;
  portfolio_value: number;
  first_access: boolean;
  super_angel: boolean;
}

export interface IAuthResponse {
  data: IInvestor;
}

export interface IAuthUser {
  email: string;
  password: string;
}

export interface IAuth {
  access_token: string;
  client: string;
  uid: string;
}

export interface IAuthState {
  auth: IAuth[];
}
