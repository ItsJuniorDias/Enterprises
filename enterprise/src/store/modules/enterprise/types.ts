export enum ActionTypes {
  getEnterpriseRequest = 'GET_ENTERPRISE_REQUEST',
  getEnterpriseSuccess = 'GET_ENTERPRISE_SUCCESS',
  getEnterpriseFailure = 'GET_ENTERPRISE_FAILURE',
  getEnterpriseHeadersSuccess = 'GET_ENTERPRISE_HEADERS_SUCCESS',
}

interface TypeEnterprise {
  id: number;
  enterprise_type_name: string;
}

export interface IHeaders {
  uid: any;
  client: any;
  'access-token': any;
}

export interface IEnterprise {
  id: number;
  email_enterprise: null;
  facebook: null;
  twitter: null;
  linkedin: null;
  phone: null;
  own_enterprise: boolean;
  enterprise_name: string;
  photo: string;
  description: string;
  city: string;
  country: string;
  value: number;
  share_price: number;
  enterprise_type: TypeEnterprise;
}

export interface IEnterpriseState {
  enterprise: IEnterprise;
}
