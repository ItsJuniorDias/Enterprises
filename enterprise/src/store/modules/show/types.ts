/* eslint-disable no-shadow */
/* eslint-disable no-unused-vars */
/* eslint-disable camelcase */
export enum ActionTypes {
  requestShow = 'REQUEST_SHOW',
  requestHeaders = 'REQUEST_HEADERS',
  successShow = 'SUCCESS_SHOW',
  failureShow = 'FAILURE_SHOW',
  resetState = 'RESET_STATE',
}

interface IEnterpriseType {
  id: number;
  enterprise_type_name: string;
}

export interface IShow {
  id: number;
  enterprise_name: string;
  description: string;
  email_enterprise: null;
  facebook: null;
  twitter: null;
  linkedin: null;
  phone: null;
  own_enterprise: null;
  photo: string;
  value: number;
  shares: number;
  share_price: number;
  own_shares: number;
  city: string;
  country: string;
  enterprise_type: IEnterpriseType;
}

export interface IHeaders {
  access_token: string;
  client: string;
  uid: string;
}

export interface IShowState {
  show: IShow;
  headers: IHeaders;
}
