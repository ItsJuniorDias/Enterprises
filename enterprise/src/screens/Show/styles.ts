import styled from 'styled-components/native';

export const SafeArea = styled.SafeAreaView`
  flex: 1;
`;

export const ContentLoading = styled.View`
  flex: 1;
`;

export const Container = styled.View`
  padding: 0 22px;
`;

export const Header = styled.View`
  width: 100%;
  height: 160px;
`;

export const Touchable = styled.TouchableOpacity`
  width: 35px;
  height: 35px;
  border-radius: 50px;
  align-items: center;
  justify-content: center;
`;

export const Row = styled.View`
  flex-direction: row;
  width: 100%;
  justify-content: space-between;
  margin-top: 24px;
`;

export const Title = styled.Text`
  font-family: 'Poppins-SemiBold';
  color: #0d253c;
  font-size: 24px;
  margin-left: 8px;
  margin-top: 24px;
`;

export const Body = styled.View`
  width: 100%;
  height: 219px;
`;

export const ImageBody = styled.Image`
  width: 100%;
  height: 219px;
  border-top-left-radius: 28px;
  border-top-right-radius: 28px;
  z-index: 1;
`;

export const ContenFilterImage = styled.View`
  width: 100%;
  background-color: rgba(0, 0, 0, 0.2);
  z-index: 5;
  height: 219px;
  border-top-left-radius: 28px;
  border-top-right-radius: 28px;
  position: absolute;
`;

export const Content = styled.View`
  width: 100%;

  padding: 0 30px;
`;

export const TitleBody = styled.Text`
  font-family: 'Poppins-SemiBold';
  font-size: 18px;
  line-height: 22px;

  color: #0d253c;
  margin-top: 32px;
`;

export const Description = styled.Text`
  font-family: 'Poppins-Regular';
  font-size: 14px;
  line-height: 24px;
  margin-top: 18px;

  color: #2d4379;
  margin-bottom: 16px;
`;

export const FloatView = styled.View`
  width: 120px;
  height: 30px;
  background-color: rgba(0, 0, 0, 0.2);
  border-radius: 16px;
  margin-top: 32px;
`;

export const FloatLocation = styled.View`
  width: 80px;
  height: 30px;
  background-color: rgba(0, 0, 0, 0.2);
  border-radius: 16px;
  margin-top: 32px;
`;

export const FloatDescription = styled.View`
  width: 160px;
  height: 30px;
  background-color: rgba(0, 0, 0, 0.2);
  border-radius: 16px;
  margin-top: 32px;
`;
