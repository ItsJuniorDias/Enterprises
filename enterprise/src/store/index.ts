import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './modules/rootReducer';

import { IAuthState } from './modules/auth/types';
import rootSaga from './modules/rootSaga';

export interface IState {
  auth: IAuthState;
}

const sagaMiddleware = createSagaMiddleware();
const middlewares = [sagaMiddleware];

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middlewares)),
);

sagaMiddleware.run(rootSaga);

export default store;
