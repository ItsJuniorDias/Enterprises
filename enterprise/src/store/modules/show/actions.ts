import { IShow, ActionTypes, IHeaders } from './types';

export function requestHeadersShow(headers: IShow) {
  return {
    type: ActionTypes.requestHeaders,
    payload: {
      headers,
    },
  };
}

export function resetState() {
  return {
    type: ActionTypes.resetState,
    enterprise: {},
  };
}

export function requestShow(headers: IHeaders, id: IShow) {
  return {
    type: ActionTypes.requestShow,
    payload: {
      id,
      headers,
    },
  };
}

export function successShow(data: IShow) {
  return {
    type: ActionTypes.successShow,
    payload: {
      data,
    },
  };
}

export function failureShow() {
  return {
    type: ActionTypes.failureShow,
    payload: {},
  };
}
