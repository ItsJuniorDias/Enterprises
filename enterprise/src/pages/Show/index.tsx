import React, { useState, useEffect } from 'react';
import {
  Image,
  ScrollView,
  Dimensions,
  BackHandler,
} from 'react-native';

import { useNavigation } from '@react-navigation/native';

import { useDispatch, useSelector } from 'react-redux';

import LinearGradient from 'react-native-linear-gradient';
import { resetState } from '../../store/modules/show/actions';

import {
  Container,
  Header,
  Touchable,
  Row,
  Title,
  Body,
  ImageBody,
  ContenFilterImage,
  Content,
  TitleBody,
  Description,
  SafeArea,
  ContentLoading,
  FloatView,
  FloatLocation,
  FloatDescription,
  ViewShimmer,
} from './styles';

import arrow from '../../assets/arrow.png';
import overflow from '../../assets/Overflow.png';

import Loading from '../../components/Loading';

const Show = () => {
  const [headerShown, setHeaderShown] = useState(false);

  const { enterprise } = useSelector(state => state.show);

  const navigation = useNavigation();
  const dispatch = useDispatch();

  const windowHeight = Dimensions.get('window').height;

  function handleBackButtonClick() {
    dispatch(resetState());
    navigation.navigate('/Home');
    return true;
  }

  useEffect(() => {
    BackHandler.addEventListener('hardwareBackPress', handleBackButtonClick);
    return () => {
      BackHandler.removeEventListener(
        'hardwareBackPress',
        handleBackButtonClick,
      );
    };
  }, []);

  const handleGoBack = () => {
    dispatch(resetState());
    navigation.goBack();
  };

  return (
    <>
      {!enterprise && (
        <>
          <ContentLoading>
            <Loading />
          </ContentLoading>
        </>
      )}

      {enterprise && (
        <>
          <SafeArea>
            <ScrollView
              onScroll={event => {
                const scrolling = event.nativeEvent.contentOffset.y;

                if (scrolling > 50) {
                  setHeaderShown(true);
                } else {
                  setHeaderShown(false);
                }
              }}
              // onScroll will be fired every 16ms
              scrollEventThrottle={16}
            >
              <Container>
                <Header>
                  <Row>
                    <Touchable onPress={() => handleGoBack()}>
                      <Image source={arrow} />
                    </Touchable>

                    <Touchable>
                      <Image source={overflow} />
                    </Touchable>
                  </Row>

                  {enterprise.enterprise_name ? (
                    <Title>{enterprise.enterprise_name}</Title>
                  ) : (
                    <>
                      <ViewShimmer
                        style={{ borderRadius: 16, marginTop: 32, height: 32 }}
                      >
                        <FloatView />
                      </ViewShimmer>
                    </>
                  )}
                </Header>
              </Container>

              <Body>
                <ImageBody
                  source={{
                    uri: `https://empresas.ioasys.com.br/${enterprise.photo}`,
                  }}
                />
                {!enterprise.photo && (
                  <ViewShimmer
                    style={{
                      width: '100%',
                      height: 219,
                      position: 'absolute',
                      borderTopLeftRadius: 28,
                      borderTopRightRadius: 28,
                    }}
                  />
                )}
                <ContenFilterImage />
              </Body>

              <Content>
                {enterprise.city ? (
                  <TitleBody>
                    {enterprise.city}, {enterprise.country}
                  </TitleBody>
                ) : (
                  <>
                    <ViewShimmer
                      style={{ borderRadius: 16, marginTop: 32, height: 24 }}
                    >
                      <FloatLocation />
                    </ViewShimmer>
                  </>
                )}
                {enterprise.description ? (
                  <Description>{enterprise.description}</Description>
                ) : (
                  <>
                    <ViewShimmer
                      style={{ borderRadius: 16, marginTop: 32, height: 24 }}
                    >
                      <FloatDescription />
                    </ViewShimmer>
                  </>
                )}
              </Content>

              <LinearGradient
                style={{
                  width: '100%',
                  height: 109,
                  position: 'absolute',
                  marginTop: windowHeight * 0.87,
                  transform: [{ translateY: headerShown ? 155 : -20 }],
                }}
                colors={[' rgba(249, 250, 255, 0.3)', '#fafbff']}
              />
            </ScrollView>
          </SafeArea>
        </>
      )}
    </>
  );
};

export default Show;
