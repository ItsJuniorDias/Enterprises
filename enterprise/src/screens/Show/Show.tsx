import React, { useEffect } from 'react';
import { Image, ScrollView, BackHandler, View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useDispatch, useSelector } from 'react-redux';
import { resetState } from '../../store/modules/show/actions';
import arrow from '../../assets/arrow.png';
import overflow from '../../assets/Overflow.png';
import { Loading } from '../../components';

import {
  Container,
  Header,
  Touchable,
  Row,
  Title,
  Body,
  ImageBody,
  ContenFilterImage,
  Content,
  TitleBody,
  Description,
  SafeArea,
  ContentLoading,
  FloatView,
  FloatLocation,
  FloatDescription,
  SkeletonPhoto,
} from './styles';

export const Show = () => {
  const { show } = useSelector((state) => state.show);

  const navigation = useNavigation();
  const dispatch = useDispatch();

  function handleBackButtonClick() {
    dispatch(resetState());

    navigation.navigate('/Home');
    return true;
  }

  useEffect(() => {
    BackHandler.addEventListener('hardwareBackPress', handleBackButtonClick);

    return () => {
      BackHandler.removeEventListener(
        'hardwareBackPress',
        handleBackButtonClick
      );
    };
  }, []);

  const handleGoBack = () => {
    dispatch(resetState());
    navigation.goBack();
  };

  return (
    <>
      {!show && (
        <>
          <ContentLoading>
            <Loading />
          </ContentLoading>
        </>
      )}

      {show && (
        <>
          <SafeArea>
            <ScrollView>
              <Container>
                <Header>
                  <Row>
                    <Touchable
                      testID="buttonTouchable_testID"
                      onPress={() => handleGoBack()}
                    >
                      <Image source={arrow} />
                    </Touchable>

                    <Touchable>
                      <Image source={overflow} />
                    </Touchable>
                  </Row>

                  {show.enterprise_name ? (
                    <Title>{show.enterprise_name}</Title>
                  ) : (
                    <FloatView />
                  )}
                </Header>
              </Container>

              <Body>
                <ImageBody
                  source={{
                    uri: `https://empresas.ioasys.com.br/${show.photo}`,
                  }}
                />

                {!show.photo && <SkeletonPhoto />}

                <ContenFilterImage />
              </Body>

              <Content>
                {show.city ? (
                  <TitleBody>
                    {show.city}, {show.country}
                  </TitleBody>
                ) : (
                  <FloatLocation />
                )}

                {show.description ? (
                  <Description>{show.description}</Description>
                ) : (
                  <FloatDescription />
                )}
              </Content>
            </ScrollView>
          </SafeArea>
        </>
      )}
    </>
  );
};
