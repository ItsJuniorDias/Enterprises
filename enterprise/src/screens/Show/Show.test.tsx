import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Show } from './Show';
import { Provider } from 'react-redux';
import produce from 'immer';
import { combineReducers, createStore } from 'redux';
import { IShowState } from 'store/modules/show/types';
import store from 'store';

jest.mock('react-native-shimmer-placeholder', () => {
  return {
    createShimmerPlaceholder: jest.fn(),
  };
});
jest.mock('react-native-linear-gradient', () => {});

const mockNavigation = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(() => ({
    goBack: mockNavigation,
    navigate: mockNavigation,
  })),
}));

export const INITIAL_STATE: IShowState = {
  show: {
    id: 2,
    email_enterprise: null,
    facebook: null,
    twitter: null,
    linkedin: null,
    phone: null,
    own_enterprise: null,
    enterprise_name: 'C6 Bank',
    photo: '',
    description:
      'O C6 Bank é uma instituição financeira com sede em São Paulo.',
    city: '',
    country: '',
    value: 0,
    share_price: 0,
    enterprise_type: null,
  },
  headers: {
    uid: '69629154-fb3b-418f-905d-5aacb7c509b0',
    client: 'SSL',
    'access-token': 'access-token',
  },
};

const show: Reducer<IShowState> = (state = INITIAL_STATE, action) => {
  return produce(state, (draft) => {
    switch (action.type) {
      default: {
        return draft;
      }
    }
  });
};

describe('Behavior Show', () => {
  function createTestStore() {
    const store = createStore(
      combineReducers({
        show,
      })
    );
    return store;
  }

  const screenRender = () => (
    <Provider store={createTestStore()}>
      <Show />
    </Provider>
  );

  it('render snapshot', () => {
    const result = render(screenRender()).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('should call function handleGoBack', () => {
    const { getByTestId } = render(screenRender());

    const touchable = getByTestId('buttonTouchable_testID');

    fireEvent.press(touchable);

    expect(mockNavigation).toBeCalled();
  });
});
