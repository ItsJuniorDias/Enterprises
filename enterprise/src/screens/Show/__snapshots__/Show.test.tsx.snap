// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior Show render snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#2D4379"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
